src/netbase.h

```diff
@@ -50,12 +55,14 @@ bool IsProxy(const CNetAddr &addr);
 bool SetNameProxy(const proxyType &addrProxy);
 bool HaveNameProxy();
 bool GetNameProxy(proxyType &nameProxyOut);
-bool LookupHost(const std::string& name, std::vector<CNetAddr>& vIP, unsigned int nMaxSolutions, bool fAllowLookup);
-bool LookupHost(const std::string& name, CNetAddr& addr, bool fAllowLookup);
-bool Lookup(const std::string& name, CService& addr, int portDefault, bool fAllowLookup);
-bool Lookup(const std::string& name, std::vector<CService>& vAddr, int portDefault, bool fAllowLookup, unsigned int nMaxSolutions);
-CService LookupNumeric(const std::string& name, int portDefault = 0);
-bool LookupSubNet(const std::string& strSubnet, CSubNet& subnet);
+
+using DNSLookupFn = std::function<std::vector<CNetAddr>(const std::string&, bool)>;
```

Comment:
This is good as it is. Maybe it would be a bit more convenient to have a single global variable of this type (resolver function) which defaults to `WrappedGetAddrInfo()` and is changed in the tests. Something like:

```cpp
DNSLookupFn g_dns_lookup = WrappedGetAddrInfo;

bool LookupHost(..., DNSLookupFn dns_lookup_function = g_dns_lookup);

// in the tests
g_dns_lookup = fuzzed_dns_lookup_function;
```

Reply:
Actually, there is more to this: using such a global would allow testing methods that call `Lookup*()` themselves. In such cases it would not be possible for the test to override the dns resolver by calling `Lookup*()` with a non-default argument.

For example `CConnman::ConnectNode()` calls `Lookup()` without passing a non-default resolver to it (of course). If we use `g_dns_lookup`, then a test could modify its value to point to a fuzzed resolver and then call `CConnman::ConnectNode()`.